### PR TITLE
Adds PagerTree Notification Provider

### PR DESCRIPTION
**Pull request by armiiller**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description

Adds [PagerTree](https://pagertree.com) Notification Provider

## Type of change

Please delete any options that are not relevant.

- User interface (UI) => Adds PagerTree Notification Provider Component (Vue)
- New feature (non-breaking change which adds functionality) => Adds PagerTree Notification Provider (Server)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

None

### CODE REVIEW

File: server/notification-providers/pagertree.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    name = "PagerTree";
+
+    /**
+     * @inheritdoc
+     */
+    async send(notification, msg, monitorJSON = null, heartbeatJSON = null) {
+        try {
+            if (heartbeatJSON == null) {
+                const title = "[Test] Uptime Kuma Alert";
+                return this.postNotification(notification, title, monitorJSON, heartbeatJSON);
```
Comment (louislam) [inline server/notification-providers/pagertree.js:17]: If `heartbeatJSON == null`, it is not for test. It is for general messages. You should send `msg`.

Comment (armiiller) [reply server/notification-providers/pagertree.js]: Thank you @louislam!

I am not sure where the interface documentation is, so I was just looking at the PagerDuty integration for reference.

I have made your requested changes.

**Comment by louislam** [comment]
Can you show screesshots of UP/DOWN/TEST notifications? As PagerTree is not available in my country I cannot test it.

image

**Comment by armiiller** [comment]
Hi @louislam  - Yes, not a problem. Please see below. Please let me know what else I can provide.

image
image
image

**Comment by pavanreddymaley** [comment]
Thanks @armiiller  @louislam  for adding the pagertree notification provider.

I have update my uptime-kuma instance from 1.19 -> 1.21.3 but I still don't see pagerTree in the notifications provider list. Is there something which I have to explicitly enable to get this working.

Thanks in advance 

Screenshot 2023-05-11 at 1 25 13 AM
Screenshot 2023-05-11 at 1 25 29 AM

**Comment by louislam** [comment]
@pavanreddymaley Thanks for the report. I think it was dropped incidentally. I just added it back: https://github.com/louislam/uptime-kuma/commit/a1adc30a898337e95e7f3c9ae1be2de1f3e8c8ef

**Comment by pavanreddymaley** [comment]
Thanks @louislam for the quick response. 
will you be releasing a new hot-fix version with the fix ??

**Comment by JimL243** [comment]
HI @pavanreddymaley & @louislam do we need to wait for that hot-fix build or is there a way to enable it otherwise? I see the pagertree.js in the container, I am not sure what the next step is. Thanks in advance.

**Comment by jacoatvatfree** [comment]
Same question from me, is there a version released that has the Pagertree notification option?

**Comment by pavanreddymaley** [comment]
@jacoatvatfree  I have updated the container image to 1.22.1 and it is showing pagertree

**Pull request merged**